lib/libzpool/uzfs_mtree.c

```diff
@@ -29,10 +29,20 @@
 
 #define	TXG_DIFF_SNAPNAME	"tsnap"
 
+#define	ADD_TO_IO_CHUNK_LIST(list, e_offset, e_len, node, count)	\
+	do {	\
+		node = umem_alloc(sizeof (*node), UMEM_NOFAIL);		\
```

Comment:
As per my understanding kmem_alloc is the right interface.

Reply:
I think umem_ related API's are for userspace memory operations. So I used umem_* API for memory allocation/free.

Reply:
the mappings are as follows: kmem -> umem -> malloc :) So in theory we could use any of these three, however for consistency reasons with the rest of the zfs code, I would recommend only the first two.

Reply:
thats right @jkryl . if we compile it for userspace then kmem and umem will lead to malloc only. But if code is meant for userspace then i think its better to use umem_ API.